File: bench/cross-language/primes_sieve/primes_sieve.go
package main

import "fmt"

func main() {
	limit := 1000000
	sieve := make([]bool, limit+1)

	sieve[0] = true
	sieve[1] = true

	for i := 2; i*i <= limit; i++ {
		if !sieve[i] {
			for j := i * i; j <= limit; j += i {
				sieve[j] = true
			}
		}
	}

	count := 0
	for i := 2; i <= limit; i++ {
		if !sieve[i] {
			count++
		}
	}

	fmt.Printf("primes_sieve(1000000): count = %d\n", count)
}

File: bench/cross-language/primes_sieve/primes_sieve.zig
const std = @import("std");

pub fn main() !void {
    const stdout_file = std.fs.File.stdout();
    var buf: [4096]u8 = undefined;
    var w = stdout_file.writer(&buf);
    const limit = 1000000;

    var sieve: [limit + 1]bool = undefined;
    @memset(&sieve, false);
    sieve[0] = true;
    sieve[1] = true;

    var i: usize = 2;
    while (i * i <= limit) : (i += 1) {
        if (!sieve[i]) {
            var j: usize = i * i;
            while (j <= limit) : (j += i) {
                sieve[j] = true;
            }
        }
    }

    var count: u32 = 0;
    for (sieve[2..]) |is_composite| {
        if (!is_composite) count += 1;
    }

    try w.interface.print("primes_sieve(1000000): count = {d}\n", .{count});
    try w.interface.flush();
}
